android/app/src/main/kotlin/coop/polypoly/polypod/FeatureFragment.kt

```diff
@@ -219,7 +222,7 @@ open class FeatureFragment : Fragment() {
         view.findViewById<TextView>(R.id.feature_title).text = title
     }
 
-    private suspend fun pickFile(type: String?): Uri? {
+    private suspend fun pickFile(type: String?): MutableMap<Value, Value>? {
```

Comment:
If possible, it should return a non-mutable map IMO.

Reply:
Can pull the translation to immutable from handlePickFile into pickFile easily

Reply:
Hm, so yeah, I don't think `ValueFactory` has any business here, actually. I think the clean thing to do would be to just return a (new) struct `ExternalFile` from here, and handle all the `ValueFactory` stuff in `PodApi`'s `handlePolyNavPickFile`.

That way we stick to our abstraction layers -> `PolyNav` operates on a typed Kotlin/Java interface, `PodApi` is responsible for serialising that using MessagePack.

Reply:
Ok makes sense. Just made that exact struct in swift - getting there :D

Reply:
That's the nice thing about implementing stuff twice (most about it isn't nice :D) we _usually_ get the design right by the time we implement it in iOS :D